### DISCUSSION TITLE
Change record data in pb_hook (edited)

### DISCUSSION THREAD

**Discussion by breakingbad45** [pocketbase/pocketbase#3716] [Q&A]
```
onRecordBeforeCreateRequest((e) => {
    console.log(e.httpContext.data)
    console.log(e.record)
    console.log(e.uploadedFiles)
}, "users", "articles")
```

can we manupulate original data

const data={
id:1001,
username:'admin'
key:"encryptedtext"
}

to this using onRecordBeforeCreateRequest

const data={
id:1001,
username:'admin'
key:"decryptedtext"
}

to this before serve the request?

**Comment by ganigeorgiev** [comment]
You can change any submitted data with `e.record.set("yourField", newValue)`.
For more details, please check https://pocketbase.io/docs/js-records/#getset-record-fields.

If that doesn't help, please be more descriptive. You've opened several discussions but without details about your use case it is really difficult to understand what you are doing and what is the expected outcome in order to provide a solution or at least suggest a workaround.

**Reply by breakingbad45** [reply]
in pb_hook can we use any 3rd party npm package?i want to when i pass payload from frontend its encrypted by any npm encryptor package and i want to make an middleware that decrypt the data and pass it to db.is it possible through pb_hook? @ganigeorgiev

**Reply by ganigeorgiev** [reply]
>  npm encryptor package...

Sorry, this doesn't really make sense to me. For external packages in `pb_hooks` please check the note in https://pocketbase.io/docs/js-overview/#loading-modules.

_Side-note: I'll be a little rude as this starts to look like a spam. I'm assuming that you and @apujayed are the same person or different people working on the same project. Both accounts in the past couple weeks have opened 10+ discussions seemingly related to the same security problem with very little and unclear descriptions, nor any consistent followups to my answers or questions (_if there was a follow up it was about something seemingly completely different as it is right now_), so I'll stop engaging with your posts as **I really don't understand what you are asking and trying to do and I really don't feel confident providing recommendations for security related questions when I don't understand the problem in the first place.**
Don't treat the repo discussions as ChatGPT. Neither me, nor probably the other users, can provide help if it is unclear what you are asking. **If you still need help, please don't open multiple discussions for the same thing and instead try to provide clear description to your problem.** It is OK to use google translate or other software if you are not a native speaker and have difficulties explaining it in English (I'm not a native speaker either and I fully understand that sometimes it could take time to elaborate on complex topics in a foreign language)._
